test/ex_dtls_test.exs

```diff
@@ -1,12 +1,15 @@
 defmodule ExDTLSTest do
-  use ExUnit.Case, async: true
+  use ExUnit.Case, async: false
 
   for impl <- [:nif, :cnode] do
     @implementation impl
     describe "#{@implementation}" do
+      setup do
+        Application.put_env(:ex_dtls, :impl, @implementation)
+      end
```

Comment:
While this shouldn't have any impact on our current tests, I think it's a good practice to reset application envs after we are done. Something like here https://elixir-lang.org/getting-started/mix-otp/config-and-releases.html#configuration

```elixir
  setup_all do
    current = Application.get_env(:kv, :routing_table)

    Application.put_env(:kv, :routing_table, [
      {?a..?m, :"foo@computer-name"},
      {?n..?z, :"bar@computer-name"}
    ])

    on_exit fn -> Application.put_env(:kv, :routing_table, current) end
  end
```